src/components/Script.jsx

```diff
@@ -1,32 +1,56 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState } from 'react';
+import { useLoaderData} from 'react-router-dom';
 import ActorScriptNav from './actorScriptNav';
 import ReadingDots from './ReadingDots';
+import Header from './Header';
 
-const Script = ({ actors, currentScript }) => {
-  const [script, setScript] = useState([]);
-  const [currentActor, setCurrentActor] = useState({});
-  const [currentCharacters, setCurrentCharacters] = useState([]);
+// Load the actor names and script titles on page load. Returns currentScript id to pass on.
+export async function loader({ params }) {
+  const currentScript = params.scriptId
+  const [loadedActors, loadedScript] = await Promise.all(
+    [getActors(), getScript(currentScript)])
 
-  // fetch the specified script
-  useEffect(() => { (async () => { // useEffect cannot take an async function. Must wrap async in regular function
-    try {
-      const response = await fetch('/script/' + currentScript);
+  return { loadedActors, loadedScript, currentScript }
+}
 
-      if (response.ok) {
-        setScript(await response.json());
-      } else {
-        console.error(`server error: ${response.body} when fetching script`);
-      }
-    } catch (error) {
-      console.error(`network error: ${error} when fetching script`);
+async function getActors() {
+  try {
+    const response = await fetch('/api/actors')
+
+    if (response.ok) {
+      return await response.json()
+    } else {
+      console.log('Server Error:', response.body);
     }
-  })();}, [currentScript]);
+  } catch (error) {
+    console.error('Network Error:', error);
+  }
+}
+
+async function getScript(scriptId) {
```

Comment:
there seems to be a lot of duplication of these functions. It may be time to do what Allison did and consolidate these into one file, or extract out the common functionality of the fetch/handle error.